src/CharityVault.sol

```diff
@@ -176,16 +178,22 @@ contract CharityVault is ERC20, Auth {
         VAULT.transfer(CHARITY, rvTokensToClaim);
     }
 
-    /// @notice returns the rvTokens owned by a user
+    /// @notice returns the rcvTokens owned by a user
     function rvTokensOwnedByUser(address user) public view returns (uint256) {
         uint256 pricePerShareNow = VAULT.exchangeRate();
 
         uint256 underlyingEarnedByUsersSinceLastExtraction = (VAULT.balanceOf(
             address(this)
         ) - (rvTokensEarnedByCharity - rvTokensClaimedByCharity)) *
             (pricePerShareNow - pricePerShareAtLastExtraction);
-        uint256 underlyingToUser = ((underlyingEarnedByUsersSinceLastExtraction *
-                this.balanceOf(user)) / totalSupply) / 100;
+
+        // Get the proportion of total interest earned for a user
+        uint256 proportionInterestEarnedByUser = (((underlyingEarnedByUsersSinceLastExtraction *
+                this.balanceOf(user)) / totalSupply) / 100);
+
+        uint256 underlyingToUser = proportionInterestEarnedByUser +
+            this.balanceOf(user);
+
```

Comment:
Had to add the user's balance of rcvTokens - I believe this logic was omitted? This way, atomic deposits and withdrawals works cc @davidlucid @JetJadeja @transmissions11